pkg/api/server/server.go

```diff
@@ -147,7 +149,27 @@ func newServer(runtime *libpod.Runtime, duration time.Duration, listener *net.Li
 	return &server, nil
 }
 
-// Serve starts responding to HTTP requests
+// If the SDNOTIFY_SOCKET is set, communicate the PID and readiness, and
+// further unset the SDNOTIFY_SOCKET to prevent containers from sending
+// messages.
+func setupSystemd() {
+	if len(os.Getenv("NOTIFY_SOCKET")) == 0 {
+		return
+	}
+	payload := fmt.Sprintf("MAINPID=%d", os.Getpid())
+	payload += "\n"
+	payload += daemon.SdNotifyReady
+	if sent, err := daemon.SdNotify(false, payload); err != nil {
+		logrus.Errorf("Error notifying systemd of Conmon PID: %s", err.Error())
+	} else if sent {
+		logrus.Debugf("Notify sent successfully")
+	}
+	if err := os.Unsetenv("NOTIFY_SOCKET"); err != nil {
```

Comment:
could we use `daemon.SdNotify(true, ...)` and avoid the `Unsetenv`?

Reply:
Good catch :+1: